internal/services/extendedlocation/extended_location_custom_location_test.go

```diff
@@ -0,0 +1,351 @@
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "22"
+    source_address_prefix      = "*"
+    destination_address_prefix = "*"
+  }
+
+  lifecycle {
+    ignore_changes = [
+      security_rule,
+    ]
+  }
+}
+
+resource "azurerm_network_interface_security_group_association" "example" {
+  network_interface_id      = azurerm_network_interface.test.id
+  network_security_group_id = azurerm_network_security_group.my_terraform_nsg.id
+}
+
+resource "azurerm_linux_virtual_machine" "test" {
+  name                            = "acctestVM-%[1]d"
+  resource_group_name             = azurerm_resource_group.test.name
+  location                        = azurerm_resource_group.test.location
+  size                            = "Standard_F2"
+  admin_username                  = "adminuser"
+  admin_password                  = "%[3]s"
+  provision_vm_agent              = false
+  allow_extension_operations      = false
+  disable_password_authentication = false
+  network_interface_ids = [
+    azurerm_network_interface.test.id,
+  ]
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+  source_image_reference {
+    publisher = "Canonical"
+    offer     = "UbuntuServer"
+    sku       = "18.04-LTS"
+    version   = "latest"
+  }
+}
+
+resource "azurerm_arc_kubernetes_cluster" "test" {
+  name                         = "acctest-akcc-%[1]d"
+  resource_group_name          = azurerm_resource_group.test.name
+  location                     = azurerm_resource_group.test.location
+  agent_public_key_certificate = "%[4]s"
+  identity {
+    type = "SystemAssigned"
+  }
+
+  %[5]s
+
+  depends_on = [
+    azurerm_linux_virtual_machine.test
+  ]
+}
+
+resource "azurerm_arc_kubernetes_cluster_extension" "test" {
+  name              = "extension-%[1]d"
+  cluster_id        = azurerm_arc_kubernetes_cluster.test.id
+  extension_type    = "microsoft.vmware"
+  release_namespace = "vmware-extension"
+
+  configuration_settings = {
+    "Microsoft.CustomLocation.ServiceAccount" = "vmware-operator"
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  depends_on = [
+    azurerm_linux_virtual_machine.test
+  ]
+}
+`, data.RandomInteger, data.Locations.Primary, credential, publicKey, provisionTemplate)
+}
+
+func (r CustomLocationResource) generateKey() (string, string, error) {
+	privateKey, err := rsa.GenerateKey(cryptoRand.Reader, 4096)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to generate RSA key")
+	}
+
+	privateKeyBytes := x509.MarshalPKCS1PrivateKey(privateKey)
+	privateKeyBlock := &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: privateKeyBytes,
+	}
+
+	privatePem := pem.EncodeToMemory(privateKeyBlock)
+	if privatePem == nil {
+		return "", "", fmt.Errorf("failed to encode pem")
+	}
+
+	return string(privatePem), base64.StdEncoding.EncodeToString(x509.MarshalPKCS1PublicKey(&privateKey.PublicKey)), nil
+}
```

Comment:
We should rather use the [`tls_private_key` Terraform resource](https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/private_key) in the test configuration to better represent a real-world configuration and also to avoid embedding redundant helper functions in test files.